src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -3639,6 +3649,710 @@ .maxstack  3
+
+            var verifierB = CompileAndVerify(compB, expectedOutput: "(1, 2, 3, 4) (1, 2, 3, 4) (10, 2, 30, 4)", verify: ExecutionConditionUtil.IsCoreClr ? Verification.Skipped : Verification.Fails);
+            // call base copy constructor B..ctor(B)
+            verifierB.VerifyIL("C..ctor(C)", @"
+{
+  // Code size       32 (0x20)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  call       ""B..ctor(B)""
+  IL_0007:  ldarg.0
+  IL_0008:  ldarg.1
+  IL_0009:  ldfld      ""object C.<P1>k__BackingField""
+  IL_000e:  stfld      ""object C.<P1>k__BackingField""
+  IL_0013:  ldarg.0
+  IL_0014:  ldarg.1
+  IL_0015:  ldfld      ""object C.<P2>k__BackingField""
+  IL_001a:  stfld      ""object C.<P2>k__BackingField""
+  IL_001f:  ret
+}");
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_WithOtherOverload()
+        {
+            var source =
+@"public record B(object N1, object N2)
+{
+    public B(C c) : this(30, 40) => throw null;
+}
+public record C(object P1, object P2) : B(3, 4)
+{
+    static void Main()
+    {
+        var c1 = new C(1, 2);
+        System.Console.Write((c1.P1, c1.P2, c1.N1, c1.N2));
+        System.Console.Write("" "");
+
+        var c2 = c1 with { P1 = 10, P2 = 20, N1 = 30, N2 = 40 };
+        System.Console.Write((c2.P1, c2.P2, c2.N1, c2.N2));
+    }
+}";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview, options: TestOptions.ReleaseExe);
+            comp.VerifyDiagnostics();
+
+            var verifier = CompileAndVerify(comp, expectedOutput: "(1, 2, 3, 4) (10, 20, 30, 40)", verify: ExecutionConditionUtil.IsCoreClr ? Verification.Skipped : Verification.Fails);
+            // call base copy constructor B..ctor(B)
+            verifier.VerifyIL("C..ctor(C)", @"
+{
+  // Code size       32 (0x20)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  call       ""B..ctor(B)""
+  IL_0007:  ldarg.0
+  IL_0008:  ldarg.1
+  IL_0009:  ldfld      ""object C.<P1>k__BackingField""
+  IL_000e:  stfld      ""object C.<P1>k__BackingField""
+  IL_0013:  ldarg.0
+  IL_0014:  ldarg.1
+  IL_0015:  ldfld      ""object C.<P2>k__BackingField""
+  IL_001a:  stfld      ""object C.<P2>k__BackingField""
+  IL_001f:  ret
+}");
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_WithObsoleteCopyConstructor()
+        {
+            var source =
+@"public record B(object N1, object N2)
+{
+    [System.Obsolete(""Obsolete"", true)]
+    public B(B b) : this(30, 40) { }
+}
+public record C(object P1, object P2) : B(3, 4) { }
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (6,1): error CS0619: 'B.B(B)' is obsolete: 'Obsolete'
```

Comment:
>// (6,1): error CS0619: 'B.B(B)' is obsolete: 'Obsolete' [](start = 16, length = 56)

I do not think this is useful. Also, I do not believe we report any obsolete diagnostics for synthesized code.